packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -481,74 +481,1755 @@ describe('NetworkController', () => {
       });
     });
 
-    [
-      NetworkType.mainnet,
-      NetworkType.goerli,
-      NetworkType.sepolia,
-      NetworkType.rpc,
-    ].forEach((networkType) => {
-      describe(`when the provider config in state contains a network type of "${networkType}"`, () => {
-        lookupNetworkTests({
-          expectedProviderConfig: buildProviderConfig({ type: networkType }),
-          initialState: {
-            providerConfig: buildProviderConfig({ type: networkType }),
-          },
-          operation: async (controller) => {
-            await controller.lookupNetwork();
-          },
+    [NetworkType.mainnet, NetworkType.goerli, NetworkType.sepolia].forEach(
+      (networkType) => {
+        describe(`when the provider config in state contains a network type of "${networkType}"`, () => {
+          describe('if the network was switched after the net_version request started but before it completed', () => {
+            it('stores the network status of the second network, not the first', async () => {
+              await withController(
+                {
+                  state: {
+                    providerConfig: buildProviderConfig({ type: networkType }),
+                    networkConfigurations: {
+                      testNetworkConfigurationId: {
+                        id: 'testNetworkConfigurationId',
+                        rpcUrl: 'https://mock-rpc-url',
+                        chainId: '0x1337',
+                        ticker: 'ABC',
+                      },
+                    },
+                  },
+                  infuraProjectId: 'some-infura-project-id',
+                },
+                async ({ controller, messenger }) => {
+                  const fakeProviders = [
+                    buildFakeProvider([
+                      // Called during provider initialization
+                      {
+                        request: {
+                          method: 'net_version',
+                        },
+                        response: SUCCESSFUL_NET_VERSION_RESPONSE,
+                      },
+                      {
+                        request: {
+                          method: 'eth_getBlockByNumber',
+                        },
+                        response: SUCCESSFUL_ETH_GET_BLOCK_BY_NUMBER_RESPONSE,
+                      },
+                      // Called via `lookupNetwork` directly
+                      {
+                        request: {
+                          method: 'net_version',
+                        },
+                        response: SUCCESSFUL_NET_VERSION_RESPONSE,
+                        beforeCompleting: () => {
```

Comment:
Nit: regarding what we discussed the other day, I think this effectively emulates an impossible scenario. But maybe the way in which it's impossible doesn't matter for this test.

If I'm understanding this correctly, this `beforeCompleting` call would run synchronously in between these network calls, but in production it's impossible for anything to run during that time frame. This could be made more realistic either by adding a delay to this call, or moving the `beforeCompleting` hook to the _second_ network call.

But it seems to make no difference to the test outcomes, at the moment at least.

Reply:
Ah, I see that the second call is covered by the other block. So adding a delay would be the only way to make this scenario more realistic then.

Reply:
Actually.. should the delay be default? Do we ever _not_ want a delay? There is always a delay for a network call in practice. Maybe we can revisit this in a later PR.

Reply:
We talked about this on video and I understand why the delay is needed. I'll make `delay: 0` the default across the board in a new PR.